### ISSUE TITLE
New deps integration doesn't handle case when project requires aliases to sync

### ISSUE THREAD

**Issue by cursive-ide** [cursive-ide/cursive#2822]
This can happen when using `:default-deps`, for example:
```clojure
{:deps    {org.clojure/core.cache nil}
 :aliases {:foo {:default-deps {org.clojure/core.cache {:mvn/version "0.6.4"}}}}}
```

This will also require #2285 to work correctly, since it will be impossible to import this project in Cursive at the moment.

**Comment by cursive-ide** [comment]
Fix released in 1.13.1-eap1 on 2023-07-24.

**Comment by cursive-ide** [comment]
Fix released in 1.13.1 on 2023-12-07.
